### ISSUE TITLE
Problems with the calculation of the loss function

### ISSUE THREAD

**Issue by jolek-jojo** [royson/metakernelgan#2]
When calculating the loss for each task，the image after downgrading the HD image once is considered **true(patch_dad),** and patch_dad goes through the generator is **fake（patch_son）**; Why is the image degraded twice as **true(gt_patch_son)** when calculating the meta-loss function, and patch_dad goes through the generator is **fake（patch_son）**? Why is the **patch_dad** used as true when calculating the task loss, and the patch_dad degraded image **gt_patch_son** as true when calculating the meta loss?Hope you can see and thank you for your answer

**Comment by royson** [comment]
hello! Thanks for taking an interest in our work. That's a fantastic question! In the **unsupervised** inner-loop (task), the **gt_patch_son** isn't available and we have to take the corresponding patch from **patch_dad** to train the discriminator with. In the **supervised** outer-loop (meta), we can simply use the **gt_patch_son** corresponding to the estimated **patch_son**. Hope this answers your question!

**Comment by jolek-jojo** [comment]
Hi,Thanks for your reply！Why is it necessary to use patch_dad for the inner loop? Why don't the inner and outer loop use gt_patch_son at the same time?Thank you again for answering my questions！

**Comment by royson** [comment]
The inner-loop is unsupervised and does not have access to the ground-truth kernel, which is used to compute gt_patch_son. In other words, the gt_patch_son is not known during task learning given an unseen image with an unseen kernel.

**Comment by jolek-jojo** [comment]
Thank you very much, I think I should get it！
